docs/hive-metrics.md

```diff
@@ -0,0 +1,108 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Hive Metrics](#hive-metrics)
+    - [Optional Metrics](#optional-metrics)
+    - [Metrics with Optional Cluster Deployment labels](#metrics-with-optional-cluster-deployment-labels)
+    - [List of all Hive metrics](#list-of-all-hive-metrics)
+      - [ClusterDeployment controller metrics](#clusterdeployment-controller-metrics)
+      - [ClusterProvision controller metrics](#clusterprovision-controller-metrics)
+      - [ClusterPool controller metrics](#clusterpool-controller-metrics)
+      - [Metrics controller metrics](#metrics-controller-metrics)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Hive Metrics
+
+Hive publishes metrics, which can help admins to monitor the Hive operations. Most of these metrics are always published, few are [optional](#metrics-with-optional-cluster-deployment-labels).
+
+### Exporting Hive Metrics
+`HiveConfig.Spec.ExportMetrics` specifies whether the operator should enable metrics for hive controllers to be extracted for the prometheus instance running on the cluster. When set to true, `openshift.io/cluster-monitoring` label is applied and  the operator deploys ServiceMonitors so that the prometheus instances can extract metrics. The operator also sets up RBAC in the TargetNamespace so that openshift prometheus in the cluster can list/access objects required to pull metrics.
+
+### Optional Metrics
+
+Some metrics pertaining to Hibernation (or PowerState) controller, are not logged by default, since that feature is not supported across all clouds.
+Admins can opt for logging such metrics via `HiveConfig.Spec.MetricsConfig`
+
+### Metrics with Optional Cluster Deployment labels
+
+Not all metrics are logged per cluster deployment name or namespace. Many metrics are logged per cluster-type, and hive allows for admins to define the labels to look for when reporting these metrics.
+
+Opt into them via `HiveConfig.Spec.MetricsConfig.AdditionalClusterDeploymentLabels`, which accepts a map of the label you would want on your metrics, to the label found on ClusterDeployment.
+For example, including `{"ocp_major_version": "hive.openshift.io/version-major"}` will cause affected metrics to include a label key ocp_major_version with the value from the `hive.openshift.io/version-major` ClusterDeployment label -- e.g. "4".
+
+Note: It is upto the cluster admins to be mindful of cardinality and ensure these labels are not too specific, like cluster id, otherwise it can negatively impact your observability system's performance
+
+### List of all Hive metrics
+
+#### ClusterDeployment controller metrics
+These metrics are logged per ClusterDeployment instance. None of these are optional.
```

Comment:
I _think_ I understand what you intend by "logged per ClusterDeployment instance", but I don't think we want to phrase it that way, as it implies we're labeling per-CD. Perhaps "These metrics are observed while processing ClusterDeployments."?